File: OrchidCore/src/main/java/com/eden/orchid/api/options/globals/ThemeGlobal.java
package com.eden.orchid.api.options.globals;

import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.api.options.TemplateGlobal;
import com.eden.orchid.api.theme.Theme;

public class ThemeGlobal implements TemplateGlobal<Theme> {

    @Override
    public String key() {
        return "theme";
    }

    @Override
    public Theme get(OrchidContext context) {
        return context.getTheme();
    }
}

File: OrchidCore/src/main/kotlin/com/eden/orchid/api/options/globals/ThemeGlobal.kt
package com.eden.orchid.api.options.globals

import com.eden.orchid.api.OrchidContext
import com.eden.orchid.api.options.TemplateGlobal
import com.eden.orchid.api.theme.Theme
import com.eden.orchid.api.theme.pages.OrchidPage

class ThemeGlobal : TemplateGlobal<Theme?> {
    override fun key() = "theme"
    override fun get(context: OrchidContext, page: OrchidPage?): Theme? = page?.theme
}
